### PR TITLE
76 flask incorporation

### LINKED ISSUE #76: Flask_Incorporation

### PR DESCRIPTION
**Pull request by mfickman**
closes #76

### CODE REVIEW

File: flagging_project/api_service/FlaggingService.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+
+#configure pymongo to integrate with flask
+def _get_connection_string(container):
+    return "mongodb://test:test@localhost:"+container.get_exposed_port(27017)
+
+def _create_flagging_mongo(container):
+    return FlaggingMongo(_get_connection_string(container))
+
+container = MongoDbContainer(MONGO_DOCKER_IMAGE)
```
Comment (akinnear) [inline flagging_project/api_service/FlaggingService.py:25]: Why is test containers code in the development class?

File: flagging_project/flag_data/FlaggingMongo.py [hunk 2]
```diff
@@ -3,6 +3,12 @@
...
 import datetime
 import pandas as pd
+from flag_data.FlaggingColumnNames import flag_name_col_name, flag_logic_col_name, \
+    referenced_flag_col_name, flag_status_col_name, flag_group_name_col_name, \
+    flag_group_flags_col_name, flag_group_status_col_name, flag_dep_dep_flags_col_name, \
+    flag_dep_flag_id_col_name, flag_dep_flag_group_id_col_name, flag_error_col_name, \
+    flag_group_error_col_name
+
 
 FLAGGING_DATABASE = 'flagging_test'
```
Comment (akinnear) [inline flagging_project/flag_data/FlaggingMongo.py:13]: A test DB in a non test class?

File: flagging_project/flag_data/FlaggingMongo.py [hunk 3]
```diff
@@ -31,176 +40,415 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+            flag_deps = None
+        return flag_deps
+
+    def delete_all_flag_dependencies(self):
+        db = get_db(self)
+        flagging_dependencies = db[FLAG_DEPENDENCIES]
+        flagging_dependencies.delete_many({})
+
+def get_db(flagging_mongo):
+    print('in get_db')
```
Comment (akinnear) [inline flagging_project/flag_data/FlaggingMongo.py:453]: Need to make sure we delete prints

File: flagging_project/front_end/FlaggingSchemaService.py [hunk 4]
```diff
@@ -7,203 +7,710 @@
...
+#                         fli_dict[attr_x] = str(value_x)
+#             else:
+#                 fli_dict[attr] = str(value)
+#     return fli_dict
 
-#A call to create a flag given a name and logic, this will return a UUID,
-# name cannot be empty if so error
+
+#FLAG
 def get_all_flags(flagging_mongo: FlaggingMongo):
```
Comment (akinnear) [inline flagging_project/front_end/FlaggingSchemaService.py:38]: I don't like including the name of the tech stack in the name. Say we switch to mysql or something else flagging_mongo would have to be changed everywhere. A better name would be flagging_dao or something like that.

**Review by akinnear — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 5]
```diff
@@ -15,6 +15,10 @@ stdlib_list = "^0.7.0"
 pymongo = "^3.11.0"
 datetime = "^4.3"
 mongomock = "^3.20.0"
+flask = "^1.1.2"
+flask_pymongo = "^2.3.0"
+requests = "^2.24.0"
+random-object-id = "^2.0.0"
```
[the remark this answers is not in the document]
Comment (mfickman) [reply pyproject.toml:21]: requests module in not necessary so I will remove from toml file.  random-object-id is for testing development

File: flagging_project/flag_data/FlaggingMongo.py (hunk 3 above)
Comment (mfickman) [reply flagging_project/flag_data/FlaggingMongo.py:453]: will do

File: flagging_project/api_service/FlaggingService.py (hunk 1 above)
Comment (mfickman) [reply flagging_project/api_service/FlaggingService.py:25]: old version of API code, will be removing.  functionality replaced my handlers.FlaggingAPI

File: flagging_project/flag_data/FlaggingMongo.py (hunk 2 above)
Comment (mfickman) [reply flagging_project/flag_data/FlaggingMongo.py:13]: good catch, was a most likely hold over

File: flagging_project/front_end/FlaggingSchemaService.py (hunk 4 above)
Comment (mfickman) [reply flagging_project/front_end/FlaggingSchemaService.py]: do I need to rename the class as well?

**Comment by mfickman** [comment]
will submit new merge request with new changes

File: pyproject.toml [hunk 6]
```diff
@@ -15,6 +15,9 @@ stdlib_list = "^0.7.0"
 pymongo = "^3.11.0"
 datetime = "^4.3"
 mongomock = "^3.20.0"
+flask = "^1.1.2"
+flask_pymongo = "^2.3.0"
+random-object-id = "^2.0.0"
```
Comment (akinnear) [inline pyproject.toml:20]: The reason I asked up the the type was that I felt like random and flask_pymongo should be under [tool.poetry.dev-dependencies]

**Review by akinnear — APPROVED** [review]
Look at the comments and fix them in the next branch

**Pull request merged**